src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1640,15 +1640,30 @@ void validateParamsType(BindingDiagnosticBag diagnostics)
                                 return;
                             }
 
-                            MethodSymbol? collectionBuilderMethod = binder.GetAndValidateCollectionBuilderMethod(syntax, (NamedTypeSymbol)Type, diagnostics, elementType: out _);
-                            if (collectionBuilderMethod is null)
-                            {
-                                return;
-                            }
+                            var targetType = (NamedTypeSymbol)Type;
+                            targetType.OriginalDefinition.HasCollectionBuilderAttribute(out TypeSymbol? builderType, out string? methodName);
+                            Debug.Assert(builderType is { });
+                            Debug.Assert(!string.IsNullOrEmpty(methodName));
 
-                            if (ContainingSymbol.ContainingSymbol is NamedTypeSymbol) // No need to check for lambdas or local function
+                            var candidateMethods = binder.GetAndValidateCollectionBuilderMethods(syntax, targetType.OriginalDefinition, builderType, methodName, diagnostics);
+                            if (candidateMethods.Any())
                             {
-                                checkIsAtLeastAsVisible(syntax, binder, collectionBuilderMethod, diagnostics);
+                                var useSiteInfo = new CompoundUseSiteInfo<AssemblySymbol>(diagnostics, ContainingAssembly);
+                                var spanType = DeclaringCompilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T).Construct(elementType);
+                                var typeArguments = targetType.GetAllTypeArguments(ref useSiteInfo);
+                                diagnostics.Add(syntax, useSiteInfo);
+                                var candidateMethodGroup = binder.BindCollectionBuilderMethodGroup(syntax, methodName, typeArguments, candidateMethods);
+                                var collectionCreation = binder.BindCollectionBuilderCreate(
+                                    syntax,
+                                    candidateMethodGroup,
+                                    spanArgument: new BoundDefaultExpression(syntax, spanType),
+                                    withElement: null, diagnostics);
+
+                                if (collectionCreation is BoundCall { Method: var collectionBuilderMethod } &&
+                                    ContainingSymbol.ContainingSymbol is NamedTypeSymbol) // No need to check for lambdas or local function
```

Comment:
I realize this is existing code, but I didn't understand how this check could come into play. We should only have instance methods from the builder type here, no?
`GetAndValidateCollectionBuilderMethods` gets the builder type (a `NamedTypeSymbol` according to `IsValidCollectionBuilderType`) then gets some of its members in `GetCollectionBuilderMethods`.

Reply:
The parameter that we're checking `ContainingSymbol.ContainingSymbol` for is the `params` parameter with a type that has a `[CollectionBuilder]` attribute, which can be a local function.

I couldn't find a test that failed when this check was removed, so I've added one.